### PR TITLE
Don't load packer at startup

### LINKED ISSUE #1427: Add impatient back
**Is your feature request related to a problem? Please describe.**
Pretty pointless to remove it in the first place to be honest
15ms without vs 21ms with impatient for me
Although it doesn't improve startuptime it does optimize real-time loading by a big margin

**Describe the solution you'd like**
In any case adding this doesn't affect `--startuptime` or `hyperfine` if you want to trick the profiler
```lua
  ["lewis6991/impatient.nvim"] = {
    event = "VimEnter",
    config = function()
      vim.defer_fn(function()
        local present, impatient = pcall(require, "impatient")
        if present then impatient.enable_profile() end
      end, 0)
    end,
  },
```
impatient will cache the lazy-loaded plugins:
image

### PR DESCRIPTION
**Pull request by nullchilly**
Closes #1427

10ms startuptime

```lua


times in msec
 clock   self+sourced   self:  sourced script
 clock   elapsed:              other lines

000.005  000.005: --- NVIM STARTING ---
000.058  000.053: event init
000.107  000.049: early init
000.130  000.023: locale set
000.152  000.021: init first window
000.318  000.166: inits 1
000.324  000.006: window checked
000.325  000.001: parsing arguments
000.564  000.034  000.034: require('vim.shared')
000.638  000.033  000.033: require('vim._meta')
000.639  000.074  000.041: require('vim._editor')
000.640  000.127  000.020: require('vim._init_packages')
000.642  000.189: init lua interpreter
000.663  000.022: expanding arguments
000.694  000.031: inits 2
000.836  000.142: init highlight
000.836  000.001: waiting for UI
001.414  000.578: done waiting for UI
001.422  000.008: init screen for UI
001.436  000.014: init default mappings
001.451  000.015: init default autocommands
001.720  000.040  000.040: sourcing /usr/local/share/nvim/runtime/ftplugin.vim
001.761  000.017  000.017: sourcing /usr/local/share/nvim/runtime/indent.vim
001.788  000.005  000.005: sourcing /usr/share/nvim/archlinux.vim
001.790  000.016  000.010: sourcing /etc/xdg/nvim/sysinit.vim
002.124  000.292  000.292: require('core')
002.272  000.099  000.099: require('core.utils')
002.452  000.159  000.159: require('core.mappings')
002.453  000.180  000.021: require('core.default_config')
003.632  001.506  001.227: require('core.options')
003.718  001.919  000.121: sourcing /tmp/nvim/init.lua
003.722  000.279: sourcing vimrc file(s)
003.913  000.014  000.014: sourcing /usr/share/vim/vimfiles/ftdetect/lf.vim
003.956  000.153  000.139: sourcing /usr/local/share/nvim/runtime/filetype.lua
003.975  000.008  000.008: sourcing /usr/local/share/nvim/runtime/filetype.vim
004.076  000.050  000.050: sourcing /usr/local/share/nvim/runtime/syntax/synload.vim
004.152  000.153  000.103: sourcing /usr/local/share/nvim/runtime/syntax/syntax.vim
004.270  000.004  000.004: require('vim.F')
004.462  000.009  000.009: sourcing /usr/local/share/nvim/runtime/plugin/gzip.vim
004.476  000.006  000.006: sourcing /usr/local/share/nvim/runtime/plugin/health.vim
004.504  000.022  000.022: sourcing /usr/local/share/nvim/runtime/plugin/man.vim
004.518  000.006  000.006: sourcing /usr/local/share/nvim/runtime/plugin/matchit.vim
004.610  000.086  000.086: sourcing /usr/local/share/nvim/runtime/plugin/matchparen.vim
004.629  000.010  000.010: sourcing /usr/local/share/nvim/runtime/plugin/netrwPlugin.vim
004.711  000.005  000.005: sourcing /tmp/nvim/rplugin.vim
004.715  000.080  000.075: sourcing /usr/local/share/nvim/runtime/plugin/rplugin.vim
004.758  000.036  000.036: sourcing /usr/local/share/nvim/runtime/plugin/shada.vim
004.772  000.006  000.006: sourcing /usr/local/share/nvim/runtime/plugin/spellfile.vim
004.787  000.008  000.008: sourcing /usr/local/share/nvim/runtime/plugin/tarPlugin.vim
004.798  000.005  000.005: sourcing /usr/local/share/nvim/runtime/plugin/tohtml.vim
004.814  000.009  000.009: sourcing /usr/local/share/nvim/runtime/plugin/tutor.vim
004.830  000.009  000.009: sourcing /usr/local/share/nvim/runtime/plugin/zipPlugin.vim
005.190  000.314  000.314: sourcing /usr/share/vim/vimfiles/plugin/fzf.vim
005.834  000.258  000.258: require('core.lazy_load')
006.052  000.182  000.182: require('base46')
006.273  000.138  000.138: require('base46.themes.onedark')
006.390  000.337  000.199: require('base46.integrations.defaults')
006.479  000.087  000.087: require('base46.glassy')
006.792  000.121  000.121: require('base46.integrations.statusline')
007.433  000.305  000.305: require('plugins.configs.others')
007.560  000.122  000.122: require('packer.load')
008.066  000.120  000.120: require('nvchad_ui.config')
008.068  000.633  000.390: require('nvchad_ui')
008.138  000.062  000.062: require('nvchad_ui.tabufline.lazyload')
008.344  003.120  001.136: sourcing /tmp/nvim/plugin/packer_compiled.lua
008.481  000.714: loading rtp plugins
008.573  000.092: loading packages
008.586  000.014: loading after plugins
008.594  000.008: inits 3
009.587  000.993: reading ShaDa
009.670  000.083: opening buffers
009.694  000.024: BufEnter autocommands
009.695  000.001: editing files in windows
009.840  000.145: VimEnter autocommands
009.842  000.002: UIEnter autocommands
010.225  000.138  000.138: sourcing /usr/local/share/nvim/runtime/autoload/provider/clipboard.vim
010.229  000.249: before starting main loop
010.457  000.068  000.068: require('nvchad_ui.statusline')
010.680  000.057  000.057: require('nvchad_ui.icons')
010.684  000.224  000.167: require('nvchad_ui.statusline.modules')
010.849  000.328: first screen update
010.850  000.002: --- NVIM STARTED ---
```

### CODE REVIEW

**Comment by siduck** [comment]
@nullchilly one thing, move this to the the core/packer.lua file 

image


Get back the boostrap function! , the main init.lua file must be smol and clean

**Review by siduck — APPROVED** [review]

File: lua/core/packer.lua [hunk 1]
```diff
@@ -1,22 +1,5 @@
 local M = {}
 
-M.bootstrap = function()
```
[the remark this answers is not in the document]
Comment (nullchilly) [reply lua/core/packer.lua:3]: @siduck It's slower by 0.1ms but okay I guess we don't strike for absolute startuptime lol

[the remark this answers is not in the document]
Comment (siduck) [reply lua/core/packer.lua:3]: > @siduck It's slower by 0.1ms but okay I guess we don't strike for absolute startuptime lol

oh how so? :thinking:

[the remark this answers is not in the document]
Comment (siduck) [reply lua/core/packer.lua:3]: ok removed last 2 commits

[the remark this answers is not in the document]
Comment (nullchilly) [reply lua/core/packer.lua:3]: @siduck 
```lua
000.005  000.005: --- NVIM STARTING ---
000.060  000.056: event init
000.115  000.055: early init
000.138  000.024: locale set
000.164  000.025: init first window
000.339  000.176: inits 1
000.346  000.006: window checked
000.347  000.001: parsing arguments
000.594  000.037  000.037: require('vim.shared')
000.670  000.035  000.035: require('vim._meta')
000.671  000.075  000.040: require('vim._editor')
000.672  000.130  000.017: require('vim._init_packages')
000.673  000.196: init lua interpreter
000.695  000.022: expanding arguments
000.729  000.034: inits 2
000.889  000.160: init highlight
000.890  000.001: waiting for UI
001.696  000.806: done waiting for UI
001.704  000.008: init screen for UI
001.724  000.020: init default mappings
001.739  000.015: init default autocommands
002.035  000.044  000.044: sourcing /usr/local/share/nvim/runtime/ftplugin.vim
002.077  000.017  000.017: sourcing /usr/local/share/nvim/runtime/indent.vim
002.110  000.008  000.008: sourcing /usr/share/nvim/archlinux.vim
002.111  000.020  000.012: sourcing /etc/xdg/nvim/sysinit.vim
002.515  000.360  000.360: require('core')
002.708  000.138  000.138: require('core.utils')
002.947  000.212  000.212: require('core.mappings')
002.949  000.239  000.026: require('core.default_config')
004.152  001.635  001.258: require('core.options')
004.666  000.150  000.150: require('core.packer')
004.718  002.595  000.450: sourcing /tmp/nvim/init.lua
004.722  000.306: sourcing vimrc file(s)
```
> 004.666  000.150  000.150: require('core.packer')

**Pull request merged**